lib/api/utilities/functions.php

```diff
@@ -81,172 +81,181 @@ function beans_remove_dir( $dir_path ) {
-			$host = preg_replace( '#' . untrailingslashit( preg_quote( $subfolder ) ) . '$#', '', $host );
+		if ( $subfolder && '/' !== $subfolder ) {
+			$pattern   = '#' . untrailingslashit( preg_quote( $subfolder ) ) . '$#';
+			$root_path = preg_replace( $pattern, '', $root_path );
+			$root_url  = preg_replace( $pattern, '', $root_url );
+		}
+
+		// If it's a multisite and not the main site, then add the site's path.
+		if ( ! is_main_site() ) {
+			$blogdetails = get_blog_details( get_current_blog_id() );
 
-			// Add the blog path for multsites.
-			if ( ! is_main_site() && ( $blogdetails = get_blog_details( get_current_blog_id() ) ) ) {
-				if ( ! ( defined( 'WP_SITEURL' ) ) || ( defined( 'WP_SITEURL' ) && WP_SITEURL == site_url() ) ) {
-					$host = untrailingslashit( $host ) . $blogdetails->path;
-				}
+			if ( $blogdetails && ( ! defined( 'WP_SITEURL' ) || ( defined( 'WP_SITEURL' ) && WP_SITEURL === site_url() ) ) ) {
+				$root_url = untrailingslashit( $root_url ) . $blogdetails->path;
 			}
 		}
 
-		$explode = beans_get( 0, explode( '/' , trailingslashit( ltrim( $subfolder, '/' ) ) ) );
-
 		// Maybe re-add tilde from host.
-		if ( false !== stripos( $explode, '~' ) ) {
-			$host = trailingslashit( $host ) . $explode;
+		$maybe_tilde = beans_get( 0, explode( '/', trailingslashit( ltrim( $subfolder, '/' ) ) ) );
+
+		if ( false !== stripos( $maybe_tilde, '~' ) ) {
+			$root_url = trailingslashit( $root_url ) . $maybe_tilde;
 		}
 	}
 
 	// Remove root if necessary.
-	if ( false !== stripos( $path, $root ) ) {
-		$path = str_replace( $root, '', $path );
+	if ( false !== stripos( $path, $root_path ) ) {
+		$path = str_replace( $root_path, '', $path );
 	} elseif ( false !== stripos( $path, beans_get( 'DOCUMENT_ROOT', $_SERVER ) ) ) {
 		$path = str_replace( beans_get( 'DOCUMENT_ROOT', $_SERVER ), '', $path );
 	}
 
-	return trailingslashit( $host ) . ltrim( $path, '/' );
-
+	return trailingslashit( $root_url ) . ltrim( $path, '/' );
 }
 
 /**
  * Convert internal url to a path.
  *
  * This function must only be used with internal urls.
  *
- * @since 1.0.0
+ * @since 1.5.0
  *
- * @param string $url Url to be converted. Accepts only internal urls.
+ * @param string $url           Url to be converted. Accepts only internal urls.
+ * @param bool   $force_rebuild Optional. Forces the rebuild of the root url and path.
  *
  * @return string Absolute path.
  */
-function beans_url_to_path( $url ) {
+function beans_url_to_path( $url, $force_rebuild = false ) {
+	static $root_path, $blogdetails;
+	$site_url = site_url();
 
-	static $root, $blogdetails;
+	if ( true === $force_rebuild ) {
+		$root_path   = '';
+		$blogdetails = '';
+	}
 
-	// Stop here if it is not an internal url.
-	if ( false === stripos( $url, parse_url( site_url(), PHP_URL_HOST ) ) ) {
-		return beans_sanitize_path( $url );
+	// Fix protocol. It isn't needed to set SSL as it is only used to parse the URL.
+	if ( ! parse_url( $url, PHP_URL_SCHEME ) ) {
```

Comment:
We must confirm (test) that this change works with IP addresses too.

Reply:
Confirmed.  New commit adds IP addresses to tests.